src/Microsoft.AspNetCore.OData/Formatter/Value/TypedEdmStructuredObject.cs

```diff
@@ -120,4 +121,27 @@ public bool TryGetPropertyValue(string propertyName, out object value)
             return helper.GetValue;
         }
     }
+
+    /// <summary>
+    /// A custom equality comparer for the property getter cache. 
+    /// </summary>
+    internal class PropertyGetterCacheEqualityComparer : IEqualityComparer<Tuple<string, Type>>
+    {
+        public bool Equals(Tuple<string, Type> x, Tuple<string, Type> y)
+        {
+            return x.Item1 == y.Item1 && x.Item2 == y.Item2;
```

Comment:
```suggestion
            return x.Equals(y);
```
`Tuple` supports Equals such that:
```csharp
using System;

var tuple1 = Tuple.Create("typeofInt", typeof(int));
var tuple2 = Tuple.Create("typeofInt", typeof(int));

Console.WriteLine(tuple1.Equals(tuple2));

// Result: True
```